### DISCUSSION TITLE
Updating RateLimit in database

### DISCUSSION THREAD

**Discussion by colbroth** [cristipufu/aspnetcore-redis-rate-limiting#169] [Q&A] [answered]
I would like to allow one to update the database value for PermitLimit and Window. I implemented being able to update database values and put a breakpoint in GetPartition to see that it picked up the new PermitLimit and Window in the SlidingRateLimiterPolicy and I do see the request got the new values. The new values are sent to GetSlidingWindowRateLimiter. But these new values do not seem to be used.

For example, I started out to allow 10 per minute. Then I updated the database to set 1 per minute. I do not see the request rejected when I issue more than 1 request after the update. 
I thought maybe the redis cache was keeping the old value so I waited until the cache was cleared but it still is allowing 10 per minute.
Is there something that I need to call to clear out the previous allowance of 10 per minute.

However if I restart my application the new value will be used so the first original settings when the application starts up seem to be the only values available leading me to believe this is cached somewhere. I would like to be able to clear that cache and set the updated values.

**Comment by cristipufu** [comment, marked as answer]
Hi @colbroth, 
The AspNet RateLimiting middleware caches the rate limiters in-memory w/o an expiration date.
This is also related to https://github.com/cristipufu/aspnetcore-redis-rate-limiting/issues/61
